rudalle/realesrgan/model.py

```diff
@@ -7,13 +7,16 @@
 
 from .rrdbnet_arch import RRDBNet
 from .utils import pad_reflect, split_image_into_overlapping_patches, stich_together, unpad_image
+from rudalle.dalle.fp16 import FP16Module
 
 
 class RealESRGAN:
-    def __init__(self, device, scale=4):
+    def __init__(self, device, scale=4, fp16=False):
         self.device = device
         self.scale = scale
         self.model = RRDBNet(num_in_ch=3, num_out_ch=3, num_feat=64, num_block=23, num_grow_ch=32, scale=scale)
+        if fp16:
```

Comment:
`self.fp16 = fp16` and after in `load_weights` before to device should use FP16Module

Reply:
and in method predict you can add:

```python
if self.fp16:
    img = torch.HalfTensor(patches / 255).permute((0, 3, 1, 2)).to(device).detach()
else:
    img = torch.FloatTensor(patches / 255).permute((0, 3, 1, 2)).to(device).detach()
```

Reply:
Thank you for the detailed feedback. Please let me know if more changes are needed.

Reply:
that's all :) 
I tried, now 4.5 gb vRAM required for super-resolution. thank you for idea! I will add in low vRAM examples.